priv/templates/boot_loader.eex

```diff
@@ -16,7 +16,7 @@ _cleanup() {
     trap - HUP INT TERM QUIT EXIT USR1 USR2
     echo ""
     echo "Shutting down.."
-    "$RELEASES_DIR/$REL_VSN/$REL_NAME.sh" rpc init stop
+    "$RELEASES_DIR/$REL_VSN/$REL_NAME.sh" <%= signal_handler %>
     wait $__pid
```

Comment:
Well I guess what I'm suggesting is that Distillery itself implement a signal handler `gen_event` which can be started as part of the release and subscribed to in order to execute custom code after a given signal - we continue to trap signals, but instead we send those signal events to the signal handler (via RPC). It's a close approximation to how people will be able to use the new OTP 20 stuff, meaning migration when 20 drops will be pretty painless, and all that would be required on Distillery's end is either just removing that stuff after some period of time, or detecting the ERTS version and conditionally enabling/disabling it.

If it's not something you can take on right now, I totally understand, but I feel like it's the right solution for this versus this new setting (which would then need to be maintained and go through deprecation, etc.).